meson.build

```
@@ -0,0 +1,139 @@
+project('pulseaudio-module-hidl', 'c',
+        version : run_command(find_program('git-version-gen'), join_paths(meson.current_source_dir(), '.tarball-version'), check: true).stdout().strip(),
+        meson_version : '>= 0.50.0',
+        default_options : [ 'c_std=gnu11' ]
+        )
+
+cc = meson.get_compiler('c')
+
+configinc = include_directories('.')
+libdir = join_paths(get_option('prefix'), get_option('libdir'))
+
+pa_c_args = ['-DHAVE_CONFIG_H']
+
+droiddevice = get_option('droid-device')
+pa_c_args += ['-DDROID_DEVICE_@0@=1'.format(droiddevice.underscorify().to_upper())]
+pa_c_args += ['-DDROID_DEVICE_STRING="@0@"'.format(droiddevice)]
+pa_c_args += ['-DLIB_AF_NAME="@0@"'.format(get_option('audioflingerglue-name'))]
+pa_c_args += ['-DLIB_AF_BASE_PATH="@0@"'.format(get_option('audioflingerglue-base'))]
+
+
+# dependencies
+droid_headers_dep = dependency('android-headers', required : true)
+hybris_dep = dependency('libhardware', version : '>= 0.1.0', required : true)
+hybris_common_dep = cc.find_library('hybris-common', required : true)
+pulsecore_dep = dependency('pulsecore', version : '>= 14.2', required : true)
+
+pa_version_str = pulsecore_dep.version()
+# For tarballs, the first split will do nothing, but for builds in git, we
+# split out suffixes when there are commits since the last tag
+# (e.g.: v11.99.1-3-gad14bdb24 -> v11.99.1)
+version_split = pa_version_str.split('-')[0].split('.')
+pa_version_major = version_split[0].split('v')[0]
+pa_version_minor = version_split[1]
+pa_version_major_minor = pa_version_major + '.' + pa_version_minor
+
+libdir = join_paths(get_option('prefix'), get_option('libdir'))
+modlibexecdir = get_option('modlibexecdir')
+if modlibexecdir == ''
+  modlibexecdir = join_paths(libdir, 'pulse-' + pa_version_major_minor, 'modules')
+endif
```

modlibexecdir is the only place libdir is used. And the only modlibexecdir is only used as the install_dir kwarg to a target.

You don't need to prepend this path with prefix, because relative paths in install_dir are always relative to prefix. You do need to manually construct the full string for cases such as configuring paths into a `configure_file()`, or passing them as `-DDEFINE` args.

```suggestion
modlibexecdir = get_option('modlibexecdir')
if modlibexecdir == ''
  modlibexecdir = join_paths(get_option('libdir'), 'pulse-' + pa_version_major_minor, 'modules')
endif
```